src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -251,6 +250,18 @@ public object GetProperyValue(object obj, string propertyName)
 
             fields.Apply(field =>
             {
+                if (objectType is IReturnsAllFields)
```

Comment:
This is just a mechanism that allows us to use interfaces, but at the same time allows for the user to request all fields on the resulting object without having to know which ones exist in the resolved implementation (so including fields that are not defined through the interface itself).

Example:
`ISomeInterface : { FieldA, FieldB }`
`Foo implements ISomeInterface : { FieldA, FieldB, FieldC }`
`Bar implements ISomeInterface : { FieldA, FieldB, FieldD, FieldE }`

For any field of type `ISomeInterface` with `IReturnsAllFields`, we essentially want to project `FieldA, FieldB, FieldC, FieldD, FieldE` (without specifying them in case they don't exist).

Thinking of it now, perhaps it's cleaner and better to just force the user to specify all "possibly" projections, instead of this "wildcard". So considering to change this. Any thoughts?